test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2375,8 +2375,18 @@ public static void setIgnoredErrorResponseCodes(Request request, RestStatus... r
         );
     }
 
+    @UpdateForV9 // remove this once we're not running REST tests against ancient 7.x versions any more
+    private static final Version VND_CONTENT_TYPES_SUPPORTED = Version.V_7_10_0; // support was added in #63071
+
+    private static XContentType randomSupportedContentType() {
+        final var oldClusterVersionString = System.getProperty("tests.old_cluster_version");
+        return oldClusterVersionString == null || Version.fromString(oldClusterVersionString).onOrAfter(VND_CONTENT_TYPES_SUPPORTED)
```

Comment:
Seems like this should be a test/cluster feature, since we are trying to move away from version checks?

Reply:
Ok I tried that in 2bf362ddaea546af393b242558d20cd9ca9d96ee.

Reply:
Yeah ok that didn't work:

`java.lang.AssertionError: Unknown historical feature rest.supports_vendor_xcontent_types: This test is running on the legacy test framework; historical features from production code will not be available. You need to port the test to the new test plugins in order to use historical features from production code. If this is a legacy feature used only in tests, you can add it to a test-only FeatureSpecification`

What is a `test-only FeatureSpecification` and how might I add one?

Reply:
@ldematte Can you help out here?

Reply:
Sure! On it

Reply:
What you did @DaveCTurner is correct, but works only with the new test framework (at least, ATM).

Also, we gave some thought about adding features that we only need to support BwC testing to production code, and we were not too fond of the idea; (historical) features can be quite short lived in production code, but need to stay much longer for some BwC tests. We rather prefer "polluting" the test code rather than production code.

So we created an additional `FeatureSpecification` to collect those, one that leaves in test code (`RestTestLegacyFeatures`), where features that are no longer needed in production (but still needed for BwC testing) end up. We might add others/split it into separate ones if it becomes too big (I think I had a draft at one point but then discarded it).

I took the liberty to add a commit that shows where I moved the `NodeFeature`; hope this helps!

Reply:
Thanks for that, I couldn't find this class myself so I added to the error message in 8a19e8f3c9b18442eaa06349c824a4c97ffb22ab to help the next person.

Reply:
Nice one, thanks for that!